File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.java
/*
 * Copyright (c) 2019 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.android.auth.callback;

import org.assertj.core.api.Assertions;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class StringAttributeInputCallbackTest {

    @Test
    public void basicTest() throws JSONException {
        JSONObject raw = new JSONObject("{\n" +
                "      \"type\": \"StringAttributeInputCallback\",\n" +
                "      \"output\": [\n" +
                "        {\n" +
                "          \"name\": \"name\",\n" +
                "          \"value\": \"sn\"\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"prompt\",\n" +
                "          \"value\": \"Last Name\"\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"required\",\n" +
                "          \"value\": false\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"policies\",\n" +
                "          \"value\": {\n" +
                "            \"policyRequirements\": [\n" +
                "              \"REQUIRED\",\n" +
                "              \"VALID_TYPE\",\n" +
                "              \"MIN_LENGTH\",\n" +
                "              \"MAX_LENGTH\"\n" +
                "            ],\n" +
                "            \"fallbackPolicies\": null,\n" +
                "            \"name\": \"sn\",\n" +
                "            \"policies\": [\n" +
                "              {\n" +
                "                \"policyRequirements\": [\n" +
                "                  \"REQUIRED\"\n" +
                "                ],\n" +
                "                \"policyId\": \"required\"\n" +
                "              },\n" +
                "              {\n" +
                "                \"policyRequirements\": [\n" +
                "                  \"VALID_TYPE\"\n" +
                "                ],\n" +
                "                \"policyId\": \"valid-type\",\n" +
                "                \"params\": {\n" +
                "                  \"types\": [\n" +
                "                    \"string\"\n" +
                "                  ]\n" +
                "                }\n" +
                "              },\n" +
                "              {\n" +
                "                \"policyId\": \"minimum-length\",\n" +
                "                \"params\": {\n" +
                "                  \"minLength\": 1\n" +
                "                },\n" +
                "                \"policyRequirements\": [\n" +
                "                  \"MIN_LENGTH\"\n" +
                "                ]\n" +
                "              },\n" +
                "              {\n" +
                "                \"policyId\": \"maximum-length\",\n" +
                "                \"params\": {\n" +
                "                  \"maxLength\": 255\n" +
                "                },\n" +
                "                \"policyRequirements\": [\n" +
                "                  \"MAX_LENGTH\"\n" +
                "                ]\n" +
                "              }\n" +
                "            ],\n" +
                "            \"conditionalPolicies\": null\n" +
                "          }\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"failedPolicies\",\n" +
                "          \"value\": []\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"validateOnly\",\n" +
                "          \"value\": true\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"value\",\n" +
                "          \"value\": \"test\"\n" +
                "        }\n" +
                "      ],\n" +
                "      \"input\": [\n" +
                "        {\n" +
                "          \"name\": \"IDToken3\",\n" +
                "          \"value\": \"\"\n" +
                "        },\n" +
                "        {\n" +
                "          \"name\": \"IDToken3validateOnly\",\n" +
                "          \"value\": true\n" +
                "        }\n" +
                "      ]\n" +
                "    }");
        StringAttributeInputCallback callback = new StringAttributeInputCallback(raw, 0);
        Assertions.assertThat(callback.getName()).isEqualTo("sn");
        Assertions.assertThat(callback.getPrompt()).isEqualTo("Last Name");
        Assertions.assertThat(callback.isRequired()).isFalse();
        Assertions.assertThat(callback.getPolicies().getString("name")).isEqualTo("sn");
        Assertions.assertThat(callback.getFailedPolicies()).isEmpty();
        Assertions.assertThat(callback.getValidateOnly()).isTrue();
        Assertions.assertThat(callback.getValue()).isEqualTo("test");
        Assertions.assertThat((boolean) callback.getInputValue(1)).isTrue();
        callback.setValidateOnly(false);
        Assertions.assertThat((boolean) callback.getInputValue(1)).isFalse();


    }
}

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/StringAttributeInputCallbackTest.kt
/*
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package org.forgerock.android.auth.callback

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.assertj.core.api.Assertions
import org.json.JSONException
import org.json.JSONObject
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class StringAttributeInputCallbackTest {
    @Test
    @Throws(JSONException::class)
    fun basicTest() {
        val raw = JSONObject("""{
      "type": "StringAttributeInputCallback",
      "output": [
        {
          "name": "name",
          "value": "sn"
        },
        {
          "name": "prompt",
          "value": "Last Name"
        },
        {
          "name": "required",
          "value": false
        },
        {
          "name": "policies",
          "value": {
            "policyRequirements": [
              "REQUIRED",
              "VALID_TYPE",
              "MIN_LENGTH",
              "MAX_LENGTH"
            ],
            "fallbackPolicies": null,
            "name": "sn",
            "policies": [
              {
                "policyRequirements": [
                  "REQUIRED"
                ],
                "policyId": "required"
              },
              {
                "policyRequirements": [
                  "VALID_TYPE"
                ],
                "policyId": "valid-type",
                "params": {
                  "types": [
                    "string"
                  ]
                }
              },
              {
                "policyId": "minimum-length",
                "params": {
                  "minLength": 1
                },
                "policyRequirements": [
                  "MIN_LENGTH"
                ]
              },
              {
                "policyId": "maximum-length",
                "params": {
                  "maxLength": 255
                },
                "policyRequirements": [
                  "MAX_LENGTH"
                ]
              }
            ],
            "conditionalPolicies": null
          }
        },
        {
          "name": "failedPolicies",
          "value": []
        },
        {
          "name": "validateOnly",
          "value": true
        },
        {
          "name": "value",
          "value": "test"
        }
      ],
      "input": [
        {
          "name": "IDToken3",
          "value": ""
        },
        {
          "name": "IDToken3validateOnly",
          "value": true
        }
      ]
    }""")
        val callback = StringAttributeInputCallback(raw, 0)
        Assertions.assertThat(callback.name).isEqualTo("sn")
        Assertions.assertThat(callback.prompt).isEqualTo("Last Name")
        Assertions.assertThat(callback.isRequired).isFalse
        Assertions.assertThat(callback.policies.getString("name")).isEqualTo("sn")
        Assertions.assertThat(callback.failedPolicies).isEmpty()
        Assertions.assertThat(callback.validateOnly).isTrue
        Assertions.assertThat(callback.value).isEqualTo("test")
        Assertions.assertThat(callback.getInputValue(1) as Boolean).isTrue
        callback.validateOnly = false
        Assertions.assertThat(callback.getInputValue(1) as Boolean).isFalse
    }
}